src/main/java/org/jsoup/helper/W3CDom.java

```diff
@@ -342,8 +344,15 @@ public void tail(org.jsoup.nodes.Node source, int depth) {
         }
 
         private void copyAttributes(org.jsoup.nodes.Node source, Element el) {
+            final Syntax syntax;
```

Comment:
I think it would be better to get the syntax from the Document's parser settings. That's what we use for every other syntax selection. And, documents don't always have doctypes. Just pass it in as an argument to this method (vs calling ownerDocument() every time, which has to walk up the parent() stack).